sshd-common/src/main/java/org/apache/sshd/common/signature/SignatureRSA.java

```diff
@@ -52,8 +52,20 @@
 
     private int verifierSignatureSize = -1;
 
+    private final String sshAlgorithmName;
+
     protected SignatureRSA(String algorithm) {
+        this(algorithm, null);
+    }
+
+    protected SignatureRSA(String algorithm, String sshAlgorithmName) {
         super(algorithm);
+        this.sshAlgorithmName = sshAlgorithmName;
+    }
```

Comment:
Let's not allow null/empty SSH algorithm name - see `ValidateUtils.checkNotNullOrEmpty`